samples/dnn/person_reid.cpp

```diff
@@ -21,187 +30,281 @@
-            for(int j = 0; j < (int)out.size().width; j++)
+            float result = 0.0;
+            for (int i = 0; i < (int)feature1.size(); i++)
             {
-                temp_feature.push_back(out.at<float>(i,j));
+                result += feature1[i] * feature2[i];
             }
-            features.push_back(normalization(temp_feature));
+            return result;
         }
-    }
-    return ;
-}
 
-static void getNames(const std::string& ImageList, std::vector<std::string>& result)
-{
-    std::ifstream img_in(ImageList);
-    std::string img_name;
-    while(img_in >> img_name)
-    {
-        result.push_back(img_name);
-    }
-    return ;
-}
+        static int getTopK(const vector<vector<float>> &queryFeatures, const vector<vector<float>> &galleryFeatures)
+        {
+            if (queryFeatures.empty() || galleryFeatures.empty())
+                return -1; // No valid index if either feature list is empty
 
-static float similarity(const std::vector<float>& feature1, const std::vector<float>& feature2)
-{
-    float result = 0.0;
-    for(int i = 0; i < (int)feature1.size(); i++)
-    {
-        result += feature1[i] * feature2[i];
-    }
-    return result;
-}
+            int bestIndex = -1;
+            float maxSimilarity = -1.0;
 
-static void getTopK(const std::vector<std::vector<float>>& queryFeatures, const std::vector<std::vector<float>>& galleryFeatures, const int& topk, std::vector<std::vector<int>>& result)
-{
-    for(int i = 0; i < (int)queryFeatures.size(); i++)
-    {
-        std::vector<float> similarityList;
-        std::vector<int> index;
-        for(int j = 0; j < (int)galleryFeatures.size(); j++)
-        {
-            similarityList.push_back(similarity(queryFeatures[i], galleryFeatures[j]));
-            index.push_back(j);
+            const vector<float> &query = queryFeatures[0];
+
+            for (int j = 0; j < (int)galleryFeatures.size(); j++)
+            {
+                float currentSimilarity = similarity(query, galleryFeatures[j]);
+                if (currentSimilarity > maxSimilarity)
+                {
+                    maxSimilarity = currentSimilarity;
+                    bestIndex = j;
+                }
+            }
+
+            return bestIndex;
         }
-        sort(index.begin(), index.end(), [&](int x,int y){return similarityList[x] > similarityList[y];});
-        std::vector<int> topk_result;
-        for(int j = 0; j < min(topk, (int)index.size()); j++)
+
+        struct MatComparator
         {
-            topk_result.push_back(index[j]);
+            bool operator()(const cv::Mat &a, const cv::Mat &b) const
+            {
+                return a.data < b.data; // This is a simple pointer comparison, not content!
+            }
+        };
+
+        map<cv::Mat, cv::Rect, MatComparator> imgDict;
+
+        static vector<cv::Mat> yoloDetector(cv::Mat &frame, Net &net, vector<cv::String> &outputLayers)
```

Comment:
verify these post processing functions are useful. Check if model can be loaded via zoo file